smarts/core/agent_manager.py

```diff
@@ -233,7 +235,9 @@ def observe(
             dones = {agent_id: True for agent_id in self.agent_ids}
             dones["__sim__"] = True
 
-        return observations, rewards, scores, dones
+        # pytype: disable=bad-return-type
+        return observations, rewards, scores, dones 
+        # pytype: enable=bad-return-type
```

Comment:
This is alarming, what is the cause here?

Reply:
Ok I have changed it, its just that the return type is quite complicated because of the `boid` condition, this could return a dictionary of observations/rewards... for one agent `{agent_id: Dict[str, Observation]}` or a single observation/reward for one agent `{agent_id: Observation}`